RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc

```diff
@@ -126,6 +126,13 @@ void SimpleNavigationSchool::linkNextForwardLayer( const BarrelDetLayer* bl,
 	fli != theRightLayers.end(); fli++) {
     if ( length < (**fli).position().z() &&
 	 radius < (**fli).specificSurface().outerRadius()) {
+      //search if there are any sovrapposition between forward layers
+      for ( FDLI fliNext = fli; fliNext != theRightLayers.end(); fliNext++) {
+        if ( (**fliNext).position().z() < (**fli).position().z() && (**fliNext).specificSurface().innerRadius() < (**fli).specificSurface().outerRadius()) {
```

Comment:
Not sure if I should introduce the Exception for both `fli` and `fliNext` or just `fli` (since this is a loop on the layers).. @slava77 what do you suggest?